### PR TITLE
comply with api key constraint

### PR DESCRIPTION
**Pull request by DaanHoogland**
## Description

moveUser only works for users without API keys which beats the (one of the two) main purpose(s) of the moveUser API.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## GitHub Issue/PRs

Fixes: #2707 

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: server/src/com/cloud/user/AccountManagerImpl.java [hunk 1]
```diff
@@ -1734,6 +1734,8 @@ public Boolean doInTransaction(TransactionStatus status) {
                 UserVO newUser = new UserVO(user);
```
Comment (rafaelweingartner) [inline server/src/com/cloud/user/AccountManagerImpl.java]: Why creating a new user entry? I mean, isn't this a matter of simply updating the accountId of the user? Then, you would not change the UUID of the "old" user entry, and you would also not need to update the keys and then mark the user as removed.

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: server/src/com/cloud/user/AccountManagerImpl.java (hunk 1 above)
Comment (DaanHoogland) [reply server/src/com/cloud/user/AccountManagerImpl.java]: that would be a nice extra feature ;)

Comment (rafaelweingartner) [reply server/src/com/cloud/user/AccountManagerImpl.java]: Ok. Then, there is one thing I still do not understand. If we want to keep the history, why are we changing the old user UUID? Shouldn't it be the opposite?

Comment (DaanHoogland) [reply server/src/com/cloud/user/AccountManagerImpl.java]: We are not changing db id which is used for foreign key relations. and we are copying the uuid to the external identity for future reference. through the new user we can then also find any 'real external' ids.

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: server/src/com/cloud/user/AccountManagerImpl.java (hunk 1 above)
Comment (rafaelweingartner) [reply server/src/com/cloud/user/AccountManagerImpl.java]: Got it. Thanks for the details.

**Comment by DaanHoogland** [comment]
not coded any tests but here is the procedure:
create accounts:
6e349308-babf-4e2a-9dde-b956bb142b9a and
c7413b70-7ef0-4dc3-9470-57719d3a354e in this example
create a user 'mover' in c7413b70-7ef0-4dc3-9470-57719d3a354e
give 'mover' keys:
(local) SBCM5> list users username=mover domainid=efdbedc6-6f38-11e8-b011-0673d0010712 listall=true
{
  "count": 1,
  "user": [
    {
      "account": "hoogland",
      "accountid": "c7413b70-7ef0-4dc3-9470-57719d3a354e",
      "accounttype": 1,
      "apikey": "PASEKfhttrjez87e9I2qsdiL95lYpvVTnVlibaMHDiJYN4KWKa_IYeFngkTP3XCp18triFmMA7jdmdmZJUTihw",
      "created": "2018-06-14T13:10:52+0000",
      "domain": "ROOT",
      "domainid": "efdbedc6-6f38-11e8-b011-0673d0010712",
      "email": "m@ov.er",
      "firstname": "Mo",
      "id": "97005683-b42a-4912-b3f7-d0bb68ee9c9e",
      "iscallerchilddomain": false,
      "isdefault": false,
      "lastname": "Ver",
      "roleid": "058540e9-6f39-11e8-b011-0673d0010712",
      "rolename": "Root Admin",
      "roletype": "Admin",
      "state": "enabled",
      "timezone": "UTC",
      "username": "mover",
      "usersource": "native"
    }
  ]
}

move user accountid=6e349308-babf-4e2a-9dde-b956bb142b9a  id=97005683-b42a-4912-b3f7-d0bb68ee9c9e 
{
  "success": "true"
}

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
The errors are mostly know to be related to other issues. @PaulAngus can we merge?

**Comment by PaulAngus** [comment]
LGTM - tested moving a user who had API keys - no issues.

..Merging

**Pull request merged**